caffe2/contrib/nccl/cuda_nccl_gpu.cc

```diff
@@ -28,8 +28,8 @@ class NCCLContext {
       // get stream priorities
       int lo_pri, hi_pri;
       CUDA_ENFORCE(cudaDeviceGetStreamPriorityRange(&lo_pri, &hi_pri));
-      CUDA_ENFORCE(cudaStreamCreateWithPriority(
-          &streams_[i], cudaStreamNonBlocking, hi_pri));
+      CUDA_ENFORCE(cudaStreamCreateWithFlags(
```

Comment:
Let's add ifdef to not affect CUDA runs.

Reply:
We are currently checking the hip API for priority settings, will get back with the analysis. In the meanwhile we are planning to check any regression for the current change, so that we can enable it in ROCM guard.

Reply:
I have updated the changes as suggested. We currently dont see any regression due to this change. In meanwhile we are checking more and will get if any updates.